From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Raine Makelainen <raine.makelainen@jolla.com>
Date: Tue, 28 Sep 2021 09:42:30 +0300
Subject: [PATCH] [sailfishos][egl] Drop swap_buffers_with_damage extension
 support. Fixes JB#55571 OMP#JOLLA-396

At least on "Adreno (TM) 508" swapping with damage triggered a hang on the
compositor thread i.e. call never returns. Conceptually I think we should
bind eglSwapBuffersWithDamage on hybris side to the eglSwapBuffers.

Co-authored-by: David Llewellyn-Jones <david.llewellyn-jones@jolla.com>
Signed-off-by: Raine Makelainen <raine.makelainen@jolla.com>
---
 gfx/gl/GLLibraryEGL.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gfx/gl/GLLibraryEGL.cpp b/gfx/gl/GLLibraryEGL.cpp
index 196004b4283b..979ed19e6c87 100644
--- a/gfx/gl/GLLibraryEGL.cpp
+++ b/gfx/gl/GLLibraryEGL.cpp
@@ -700,6 +700,8 @@ EglDisplay::EglDisplay(const PrivateUseOnly&, GLLibraryEGL& lib,
   MarkExtensions(rawExtString, shouldDumpExts, "display", sEGLExtensionNames,
                  &mAvailableExtensions);
 
+  MarkExtensionUnsupported(EGLExtension::EXT_swap_buffers_with_damage);
+  MarkExtensionUnsupported(EGLExtension::KHR_swap_buffers_with_damage);
   // -
 
   if (!HasKHRImageBase()) {
